src/core/core.scale.js

```diff
@@ -829,30 +837,38 @@ module.exports = function(Chart) {
 				context.restore();
 			}
 
-			if (gridLines.drawBorder) {
-				// Draw the line at the edge of the axis
-				context.lineWidth = helpers.valueAtIndexOrDefault(gridLines.lineWidth, 0);
-				context.strokeStyle = helpers.valueAtIndexOrDefault(gridLines.color, 0);
-				var x1 = me.left;
-				var x2 = me.right;
-				var y1 = me.top;
-				var y2 = me.bottom;
+			// gridLines.drawBorder is deprecated
+			if (gridLines.drawBorder && options.borderColor !== null && options.borderWidth !== 0 && options.borderWidth !== null) {
```

Comment:
we discussed that `borderWidth === 0` should disable the border and it should not be drawn. Thats why I check for it here. Its also checked while collecting the gridLines to determine if it should be added to the hash